lib/src/core/verifier/mod.rs

```diff
@@ -0,0 +1,65 @@
+use crate::core::*;
+use anyhow::Result;
+use async_trait::async_trait;
+use std::path::PathBuf;
+
+pub mod policy;
+mod sample;
+mod sgx;
+mod tdx;
+
+#[async_trait]
+pub trait Verifier {
+    async fn evaluate(&self, evidence: &Evidence) -> Result<AttestationResults>;
+    fn opa_files_path(&self, file: Files, tee: TEEs) -> Result<PathBuf>;
+}
+
+#[async_trait]
+pub trait TEETraits {
+    async fn evaluate(&self, evidence: &Evidence) -> Result<AttestationResults>;
+    fn opa_policy_path(&self) -> Result<PathBuf>;
+    fn opa_reference_data_path(&self) -> Result<PathBuf>;
+}
+
+#[derive(Debug)]
+pub struct Verifiers {
+    pub tdx: tdx::Tdx,
+    pub sgx: sgx::Sgx,
+    pub sample: sample::Sample,
+}
```

Comment:
Yes, I will reconstruct the codes according to your comments. And I also rethink the `evaluation()` and concluded that let this Crate to manage and store some persistent files(such as policy and reference data) is inapporiate. So I switch the function interface into this:
```RUST
    async fn evaluate(
        &self,
        evidence: &Evidence,
        policy: Option<String>,
        reference_data: Option<String>,
    ) -> Result<AttestationResults> ;
```
So I it can receive the external policy and reference data to execute evaluation. If the parameters are empty, it will use the   default policy and reference data.

Reply:
Please let me know when the updated implementation is ready for a follow up review.

Reply:
Yes, I have updated this PR according to all these review comments. And I also found and fixed two bugs during some development working which depends on this PR's Crate.
@sameo @fitzthum @dubek Please help to review this PR again.